sklearn/sklearn-template/template/scripts/train.sh

```diff
@@ -0,0 +1,75 @@
+#!/usr/bin/env bash
+
+# Copyright 2018 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Convenience script for running ML training jobs.
+#
+# Prerequisites:
+#   - Google Cloud SDK
+#
+# Globals:
+#   PROJECT_ID: Google Cloud project to use.
+#   BUCKET_ID: Google Cloud Storage bucket to store output.
```

Comment:
I think there is no need to make BUCKET_ID as global. Since BUCKET_ID is not commonly configured, it's better to make them aware in stead of using environment.

Reply:
Are you thinking of defining `BUCKET_ID` in the script itself or as a script parameter?

For the first case, having the `BUCKET_ID` specified within the script might make the code less portable. Also for template development, whenever we want to check in changes to `train.sh`, we'd have to make sure that `BUCKET_ID=''` so that we don't commit any reference to an actual GCP bucket.

For the second case, `BUCKET_ID` won't probably change much during the model development and it would be a bit cumbersome to pass it to the script every time the script needs to be run.   

Alternatively, I was thinking we could have a train config file that could define all the necessary params, and then create a Python script that just reads from that config file and runs CMLE via Python API calls. The config file will have the same caveats as the first case specified above, but would probably easier to maintain (e.g. remove the actual param values before checking in changes). I thought of leaving this as future work and for now stick to Bash scripts for CMLE execution until the template is ready to be merged to the fork's origin.